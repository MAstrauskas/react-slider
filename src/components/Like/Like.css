/* Heart like button */
.like-heart {
    position: absolute;
    cursor: pointer;
    margin-left: 18px;
    width: 1%;
    height: 9%;
    bottom: 16px;
}

.like-heart img {
    position: absolute;
    width: 50%;
}
