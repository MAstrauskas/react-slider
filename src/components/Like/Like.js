import React, { Component } from "react";
import HeartIcon from "../../assets/icons/like.svg";
import HeartActiveIcon from "../../assets/icons/like-active.svg";

import "./Like.css";

class Like extends Component {
    constructor(props) {
        super(props);

        this.state = {
            isLiked: this.props.is_liked,
            heartIcon: HeartIcon
        }
    }

    componentDidMount() {
        this.likeCard();
    }

    /**
     * Function that triggers the action
     * of liking and un-liking.
     */
    likeCard = () => {
        if (!this.state.isLiked) {
            this.setState(() => {
                return {
                    isLiked: true,
                    heartIcon: HeartIcon
                }

            });
        } else {
            this.setState(() => {
                return {
                    isLiked: false,
                    heartIcon: HeartActiveIcon,
                }
            })
        }
    };

    updateLike(isLiked) {
            fetch(`http://127.0.0.1:3001/cards/${this.props.id}`, {
                headers: {"Content-Type": "application/json; charset=utf-8"},
                method: "PATCH",
                body: JSON.stringify({
                    is_liked: isLiked,
                })
            })
                .then(res => res.json())
                .then(like => like)
                .catch(err => err);
    }

    render() {
        return (
            <div className="like-heart" onClick={() => {this.likeCard(); this.updateLike(this.state.isLiked);} }>
                <img src={this.state.heartIcon} alt="Green Heart Button" />
            </div>
        )
    }
}

export default Like;