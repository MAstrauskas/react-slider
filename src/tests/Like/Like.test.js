import React from "react";
import { create } from "react-test-renderer";
import Like from "../../components/Like/Like";

describe("Like Component", () => {
    test("it matches the snapshot", () => {
        const component = create(<Like/>);
        expect(component.toJSON()).toMatchSnapshot();
    })
});