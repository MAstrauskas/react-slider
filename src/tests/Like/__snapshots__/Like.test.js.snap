// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Like Component it matches the snapshot 1`] = `
<div
  className="like-heart"
  onClick={[Function]}
>
  <img
    alt="Green Heart Button"
    src="like.svg"
  />
</div>
`;
